src/Program_UI.c

```diff
@@ -1,6 +1,190 @@
+            fprintf(file, "%d\n", profiles[i].answers.societyInterest);
+            fprintf(file, "%d\n", profiles[i].answers.scientificInterest);
+            fprintf(file, "%d\n", profiles[i].answers.businessEcon);
+            fprintf(file, "%d\n", profiles[i].answers.mathFascination);
+            fprintf(file, "%d\n", profiles[i].answers.careFulfillment);
+            fprintf(file, "%d\n", profiles[i].answers.languages);
+            fprintf(file, "%d\n", profiles[i].answers.jobPriority);
+            fprintf(file, "%d\n", profiles[i].answers.salaryPriority);
+        }
+        fclose(file);
+    }
+}
+
+void loadProfiles(const char *fileName, UserProfile *profiles, int *numProfiles) {
+    FILE *file = fopen(fileName, "r");
+    if (file != NULL) {
+        fscanf(file, "%d", numProfiles);
+
+        for (int i = 0; i < *numProfiles && i < 25; i++) {
+            fscanf(file, "%s", profiles[i].name);
+            fscanf(file, "%f", &profiles[i].grades.grade);
+            fscanf(file, "%d", &profiles[i].answers.passedMathA);
+            fscanf(file, "%d", &profiles[i].answers.locationPreference);
+            fscanf(file, "%d", &profiles[i].answers.creativeInterest);
+            fscanf(file, "%d", &profiles[i].answers.technicalInterest);
+            fscanf(file, "%d", &profiles[i].answers.teamworkPreference);
+            fscanf(file, "%d", &profiles[i].answers.itInterest);
+            fscanf(file, "%d", &profiles[i].answers.independencePreference);
+            fscanf(file, "%d", &profiles[i].answers.practicalExperienceValue);
+            fscanf(file, "%d", &profiles[i].answers.societyInterest);
+            fscanf(file, "%d", &profiles[i].answers.scientificInterest);
+            fscanf(file, "%d", &profiles[i].answers.businessEcon);
+            fscanf(file, "%d", &profiles[i].answers.mathFascination);
+            fscanf(file, "%d", &profiles[i].answers.careFulfillment);
+            fscanf(file, "%d", &profiles[i].answers.languages);
+            fscanf(file, "%d", &profiles[i].answers.jobPriority);
+            fscanf(file, "%d", &profiles[i].answers.salaryPriority);
+        }
+        fclose(file);
+    }
 }
+
+// Function to display user profiles
+void displayProfiles(UserProfile *profiles, int numProfiles) {
+    printf("\n----- User Profiles -----\n");
+
+    // Prompt the user for a profile name
+    char searchName[50];
+    printf("Enter the name of the profile to display: ");
+    scanf("%s", searchName);
+
+    int found = 0;
+
+    for (int i = 0; i < numProfiles; i++) {
+        if (strcmp(profiles[i].name, searchName) == 0) {
+            printf("Profile Name: %s\n", profiles[i].name);
+            printf("Grade: %.2f\n", profiles[i].grades.grade);
+
+            printf("Passed Mathematics A: %s\n", profiles[i].answers.passedMathA ? "Yes" : "No");
+            printf("Location Preference: %d\n", profiles[i].answers.locationPreference);
```

Comment:
instead of using profiles[i].answers.{something} consider using choice_vector.Answer[{some int}}